Content.Server/Nyanotrasen/Mail/MailSystem.cs

```diff
@@ -194,26 +182,25 @@ private void OnAfterInteractUsing(EntityUid uid, MailComponent component, AfterI
             if (!args.CanReach || !component.IsLocked)
                 return;
 
-            if (!TryComp<AccessReaderComponent>(uid, out var access))
+            if (!HasComp<AccessReaderComponent>(uid))
                 return;
 
             IdCardComponent? idCard = null; // We need an ID card.
 
-            if (HasComp<PdaComponent>(args.Used)) /// Can we find it in a PDA if the user is using that?
+            if (HasComp<PdaComponent>(args.Used)) // Can we find it in a PDA if the user is using that?
             {
-                _idCardSystem.TryGetIdCard(args.Used, out var pdaID);
-                idCard = pdaID;
+                _idCardSystem.TryGetIdCard(args.Used, out var pdaId);
+                idCard = pdaId;
             }
-
-            if (HasComp<IdCardComponent>(args.Used)) /// Or are they using an id card directly?
+            else if (HasComp<IdCardComponent>(args.Used)) // Or are they using an id card directly?
```

Comment:
TryGetIdCard could fail - if you had an empty PDA that itself _was_ an ID card (somehow), you'd get left without an ID card with this.

Maybe replace L195 with this if you want to keep the priority the same?
`if (idCard == null && HasComp<IdCardComponent>(args.Used)) // Or are they using an id card directly?`